packages/docusaurus-theme-classic/src/theme/DocItem/index.js

```diff
@@ -11,16 +11,26 @@ import Head from '@docusaurus/Head';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import useBaseUrl from '@docusaurus/useBaseUrl';
 import DocPaginator from '@theme/DocPaginator';
+import useTableOfContentHighlight from '@theme/hooks/useTableOfContentHighlight';
 
 import styles from './styles.module.css';
 
 function Headings({headings, isChild}) {
+  const LINK_CLASS_NAME = 'contents__link';
+  const TOP_OFFSET = 100;
+
+  useTableOfContentHighlight(
+    LINK_CLASS_NAME,
+    'contents__link--active',
```

Comment:
Since this is BEM, the classes are dependent, which means you can do so:

```suggestion
    `${LINK_CLASS_NAME}--active`,
```

Reply:
I'd prefer not to do concat CSS classnames and instead declare another constant called `LINK_ACTIVE_CLASS_NAME` or something. Concatenating class names makes the code harder to statically analyze.

I find the original way acceptable too.